### PR TITLE
[OSDOCS-3498]ROSA STS docs: least-privilege user IAM for CLI user

### PR DESCRIPTION
**Pull request by mletalie**
Version(s):

4.15+
Issue:

https://issues.redhat.com/browse/OSDOCS-3498
Link to docs preview:

https://73430--ocpdocs-pr.netlify.app/openshift-rosa/latest/cli_reference/rosa_cli/rosa-cli-permission-examples
QE review:
- [x] QE has approved this change.

Additional information:

### CODE REVIEW

File: modules/rosa-cli-no-permissions-required.adoc [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+== rosa list oidc-config
+
+.Input
+[source,terminal]
+----
+$ rosa list oidc-config
+----
+
+[id="rosa-list-idps-cluster_{context}"]
+== rosa list idps
```
Comment (jneczypor) [inline modules/rosa-cli-no-permissions-required.adoc:76]:
```suggestion
== List identity providers
```

File: modules/rosa-cli-no-permissions-required.adoc [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+.Input
+[source,terminal]
+----
+$ rosa list idps -c <cluster-name>
+----
+
+[id="rosa-list-ingresses_{context}"]
+== rosa list ingresses
```
Comment (jneczypor) [inline modules/rosa-cli-no-permissions-required.adoc:85]:
```suggestion
== List ingresses
```

File: modules/rosa-cli-no-permissions-required.adoc [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+.Input
+[source,terminal]
+----
+$ rosa list ingresses -c <cluster-name>
+----
+
+[id="rosa-list-machinepools-cluster-name_{context}"]
+== rosa list machinepools
```
Comment (jneczypor) [inline modules/rosa-cli-no-permissions-required.adoc:94]:
```suggestion
== List machine pools
```

**Review by jneczypor — COMMENTED** [review]
This is a great start. My comments are mostly the same re-write over and over for each command. Make sure you run everything through Vale and check all your headings for capitalization before this goes to peer review!

**Comment by mletalie** [comment]
> This is a great start. My comments are mostly the same re-write over and over for each command. Make sure you run everything through Vale and check all your headings for capitalization before this goes to peer review!

Thank you, will do! Need to get Vale into my process.....starting now!

**Comment by mletalie** [comment]
Hello @arendej, 
     Will you please take another look at this PR when you get a moment? 
In regard to this comment: https://redhat-internal.slack.com/archives/C0308P3FC7R/p1711986179302469?thread_ts=1711983449.564719&cid=C0308P3FC7R

I  spoke to Sadique to get further clarity about this and his thought was: "The discussion is only about what permissions are needed for AWS access key and secret key. So there is no need to specifically mention those set of commands need a Rosa token." 

Let me know if that works for you and/or you see any other changes you would like made.

**Review by arendej — APPROVED** [review]
LGTM

File: modules/rosa-cli-hcp-classic-examples.adoc [hunk 4]
```diff
@@ -0,0 +1,254 @@
...
+
+:_mod-docs-content-type: REFERENCE
+[id="rosa-cli-hcp-classic-examples_{context}"]
+= Least privilege permissions for common ROSA CLI commands
+
+The following required minimum permissions for the listed ROSA CLI commands are applicable for hosted control plane (HCP) and Classic clusters.
+
+[id="rosa-create-OIDC-providers-hcp-classic_{context}"]
+== Create an OpenID Connect (OIDC) provider
+Run the following command with the specified permissions to create your OIDC provider by using `auto` mode.
```
Comment (sadiquepp) [inline modules/rosa-cli-hcp-classic-examples.adoc:13]: Run the following command with the specified permissions to create Managed OIDC provider by using `auto` mode.

File: modules/rosa-cli-hcp-classic-examples.adoc [hunk 5]
```diff
@@ -0,0 +1,254 @@
...
+            "Resource": "*"
+        }
+    ]
+}
+----
+
+[id="rosa-delete-oidc-config-hcp-classic_{context}"]
+== Delete your OIDC configuration
+
+Run the following command with the specified permissions to delete your OIDC configuration by using `auto` mode.
```
Comment (sadiquepp) [inline modules/rosa-cli-hcp-classic-examples.adoc:196]: Run the following command with the specified permissions to delete Managed OIDC configuration by using `auto` mode.

File: modules/rosa-cli-hcp-examples.adoc [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+
+[id="rosa-create-account-operator-roles-hcp_{context}"]
+== Create your account roles and Operator roles
+
+Run the following command with the specified permissions to create account and Operator roles by using `auto` mode.
+
+.Input
+[source,terminal]
+----
+$ rosa create account-roles --mode auto
```
Comment (sadiquepp) [inline modules/rosa-cli-hcp-examples.adoc:53]: rosa create account-roles --mode auto --hosted-cp

File: modules/rosa-cli-classic-examples.adoc [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+
+[id="rosa-create-account-operator-roles-classic_{context}"]
+== Create account roles and Operator roles
+
+Run the following command with the specified permissions to create account and Operator roles in `auto' mode.
+
+.Input
+[source,terminal]
+----
+$ rosa create account-roles --mode auto
```
Comment (sadiquepp) [inline modules/rosa-cli-classic-examples.adoc:51]: $ rosa create account-roles --mode auto --classic

**Comment by yuwang-RH** [comment]
lgmt

**Comment by sadiquepp** [comment]
lgmt

**Review by EricPonvelle — COMMENTED** [review]
/lgtm

**Comment by mletalie** [comment]
> /lgtm
Thanks!

**Pull request merged**